idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtClassMember.kt

```diff
@@ -0,0 +1,269 @@
+): KtCallableDeclaration = with(ktClassMember) {
+    val bodyType = when {
+        targetClass?.hasExpectModifier() == true -> BodyType.NO_BODY
+        (symbol as? KtPossibleExtensionSymbol)?.isExtension == true && mode == MemberGenerateMode.OVERRIDE -> BodyType.FROM_TEMPLATE
+        else -> bodyType
+    }
+
+    val renderOptions = when (mode) {
+        MemberGenerateMode.OVERRIDE -> RenderOptions.overrideRenderOptions
+        MemberGenerateMode.ACTUAL -> RenderOptions.actualRenderOptions
+        MemberGenerateMode.EXPECT -> RenderOptions.expectRenderOptions
+    }
+    if (preferConstructorParameter && symbol is KtPropertySymbol) {
+        return generateConstructorParameter(project, symbol, renderOptions, mode == MemberGenerateMode.OVERRIDE)
+    }
+
+
+    val newMember: KtCallableDeclaration = when (symbol) {
+        is KtFunctionSymbol -> generateFunction(project, symbol, renderOptions, bodyType, mode == MemberGenerateMode.OVERRIDE)
+        is KtPropertySymbol -> generateProperty(project, symbol, renderOptions, bodyType, mode == MemberGenerateMode.OVERRIDE)
+        else -> error("Unknown member to override: $symbol")
+    }
+
+    when (mode) {
+        MemberGenerateMode.ACTUAL -> newMember.addModifier(KtTokens.ACTUAL_KEYWORD)
+        MemberGenerateMode.EXPECT -> if (targetClass == null) {
+            newMember.addModifier(KtTokens.EXPECT_KEYWORD)
+        }
+        MemberGenerateMode.OVERRIDE -> {
+            // TODO: add `actual` keyword to the generated member if the target class has `actual` and the generated member corresponds to
+            //  an `expect` member.
+        }
+    }
+
+    if (copyDoc) {
+        val kDoc = when (val originalOverriddenPsi = symbol.originalOverriddenSymbol?.psi) {
+            is KtDeclaration ->
+                findDocComment(originalOverriddenPsi)
+            is PsiDocCommentOwner -> {
+                val kDocText = originalOverriddenPsi.docComment?.let { IdeaDocCommentConverter.convertDocComment(it) }
+                if (kDocText.isNullOrEmpty()) null else KDocElementFactory(project).createKDocFromText(kDocText)
+            }
+            else -> null
+        }
+        if (kDoc != null) {
+            newMember.addAfter(kDoc, null)
+        }
+    }
+
+    return newMember
+}
+
+private fun KtAnalysisSession.generateConstructorParameter(
+    project: Project,
+    symbol: KtCallableSymbol,
+    renderOptions: KtDeclarationRendererOptions,
+    isOverride: Boolean,
+): KtCallableDeclaration {
+    return KtPsiFactory(project).createParameter(symbol.render(renderOptions.copy(forceOverrideModifier = isOverride)))
+}
+
+private fun KtAnalysisSession.generateFunction(
+    project: Project,
+    symbol: KtFunctionSymbol,
+    renderOptions: KtDeclarationRendererOptions,
+    bodyType: BodyType,
+    isOverride: Boolean,
+): KtCallableDeclaration {
+    val returnType = symbol.annotatedType.type
+    val returnsNotUnit = !returnType.isUnit
```

Comment:
This is confusing to read when it's negated below. Maybe you can just do `returnsUnit`?

Reply:
I have no idea why I did that! 🤦